scipy/stats/_continuous_distns.py

```diff
@@ -906,8 +906,43 @@ def _cdf(self, x, a, b):
             f2=lambda x_, a_, b_: beta._cdf(x_/(1+x_), a_, b_))
 
     def _ppf(self, p, a, b):
-        r = sc.betaincinv(a, b, p)
-        return r / (1 - r)
+        # _ppf_direct is the direct way to compute the ppf (except for the
+        # additional safety mechanism described in the comment below)
+        # f2 is based on the inversion of the expression using beta._sf in
+        # self.cdf, see comments there.
+        # note that p=1 has to be handled separately to avoid division by 0
+
+        def _ppf_direct(p_, a_, b_):
+            r = stats.beta.ppf(p_, a_, b_)
```

Comment:
Whatever you decide with the `_lazywhere`s, you can use `_ppf`, etc.

I am fine with not using `special.betaincinv` directly, though. It is a bit faster for the fastest problems, negligibly faster for the largest problems, and always harder to read (such that it results in mistakes - see gh-17797).

Reply:
oops, I missed this one... once we decide if we keep lazywhere or not, we can change it.